test/windows/WsladiagTests.cpp

```diff
@@ -0,0 +1,166 @@
+/*++
+
+Copyright (c) Microsoft. All rights reserved.
+
+Module Name:
+
+    WsladiagTests.cpp
+
+Abstract:
+
+    This file contains smoke tests for wsladiag.
+
+--*/
+
+#include "precomp.h"
+#include "Common.h"
+#include <format>
+
+namespace WsladiagTests {
+class WsladiagTests
+{
+    WSL_TEST_CLASS(WsladiagTests)
+
+    // Initialize the tests
+    TEST_CLASS_SETUP(TestClassSetup)
+    {
+        VERIFY_ARE_EQUAL(LxsstuInitialize(FALSE), TRUE);
+        return true;
+    }
+
+    TEST_CLASS_CLEANUP(TestClassCleanup)
+    {
+        LxsstuUninitialize(FALSE);
+        return true;
+    }
+    // Test that wsladiag list command shows either sessions or "no sessions" message
+    TEST_METHOD(List_ShowsSessionsOrNoSessions)
+    {
+        auto [out, err, code] = RunWsladiag(L"list");
+        VERIFY_ARE_EQUAL(0, code);
+        VERIFY_ARE_EQUAL(L"", err);
+
+        ValidateListOutput(out);
+    }
+
+    // Test that wsladiag --help shows usage information
+    TEST_METHOD(Help_ShowsUsage)
+    {
+        auto [out, err, code] = RunWsladiag(L"--help");
+        VERIFY_ARE_EQUAL(0, code);
+        VERIFY_ARE_EQUAL(L"", out);
+        ValidateUsage(err);
+    }
+
+    // Test that wsladiag with no arguments shows usage information
+    TEST_METHOD(EmptyCommand_ShowsUsage)
+    {
+        auto [out, err, code] = RunWsladiag(L"");
+        VERIFY_ARE_EQUAL(0, code);
+        VERIFY_ARE_EQUAL(L"", out);
+        ValidateUsage(err);
+    }
+
+    // Test that -h and --help flags produce identical output
+    TEST_METHOD(Help_ShortAndLongFlags_Match)
+    {
+        auto [outH, errH, codeH] = RunWsladiag(L"-h");
+        auto [outLong, errLong, codeLong] = RunWsladiag(L"--help");
+
+        VERIFY_ARE_EQUAL(0, codeH);
+        VERIFY_ARE_EQUAL(0, codeLong);
+
+        VERIFY_ARE_EQUAL(L"", outH);
+        VERIFY_ARE_EQUAL(L"", outLong);
+
+        VERIFY_ARE_EQUAL(errH, errLong);
+        ValidateUsage(errH);
+    }
+
+    // Test that unknown commands show error message and usage
+    TEST_METHOD(UnknownCommand_ShowsUsage)
+    {
+        auto [out, err, code] = RunWsladiag(L"blah");
+        VERIFY_ARE_NOT_EQUAL(0, code);
```

Comment:
I'd recommend validating that the error code is 1 explicitly, since that's what we expect